src/main/java/org/jooq/lambda/SequentialSpliterator.java

```diff
@@ -0,0 +1,39 @@
+package org.jooq.lambda;
+
+import java.util.Spliterator;
+
+/**
+ * Special subinterface of <code>Spliterator</code> that provides default implementations
+ * for all the methods that are not relevant to sequential traversal.
+ *
+ * @see java.util.stream.BaseStream#sequential
+ *
+ * @author Tomasz Linkowski
+ */
+@FunctionalInterface
+public interface SequentialSpliterator<T> extends Spliterator<T> {
```

Comment:
Hmm, interesting - good idea. Perhaps we should put that in a separate issue, though. I'll decide later, when it's clear if this type really needs to be public